services/workflows-service/src/transaction/transaction.service.ts

```diff
@@ -37,8 +40,16 @@ export class TransactionService {
           correlationId: transaction.transactionCorrelationId,
         });
       } catch (error) {
+        let errorToLog: Error = new Error('Unknown error', { cause: error });
+
+        if (error instanceof Prisma.PrismaClientKnownRequestError && error.code === 'P2002') {
```

Comment:
you can use some prisma util functions like 'isFkConstraintError' which located at `services/workflows-service/src/prisma/prisma.util.ts`

Reply:
When will this occur?